Generators/AutoEquality.cs

```diff
@@ -159,24 +162,20 @@ void AddEquals()
 
                 for (var i = 0; i < memberInfoList.Count; i++)
                 {
-                    var current = memberInfoList[i];
-                    if (current.UseOperator)
-                    {
-                        builder.Append($"{indent.Value}{current.Name} == other.{current.Name}");
-                    }
-                    else
-                    {
-                        builder.Append($"{indent.Value}EqualityComparer<{current.TypeName}>.Default.Equals({current.Name}, other.{current.Name})");
-                    }
-
-                    if (i + 1 < memberInfoList.Count)
-                    {
-                        builder.Append(" &&");
-                    }
-                    else
+                    var (name, typeName, useOperator) = memberInfoList[i];
+                    var line = (
+                        useOperator,
+                        isString: typeName.Equals("string", StringComparison.OrdinalIgnoreCase)) switch
```

Comment:
This is not a rigorous check for whether we are dealing with the type `string`. Depending on the name in general will either allow incorrect cases or miss valid ones. 

https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html

The best way to do this is to bind the type and then check for the `SpecialType.String`

Reply:
Ah, perfect - I was curious about that. I will update it when I'm back at my machine. I could pluck that out when doing the `memberList` build up. Thanks 🤘🏼